src/foundry/common/utils/bitmask.d.mts

```diff
@@ -5,46 +5,12 @@ export default class BitMask extends Number {
    */
   constructor(states?: Record<string, boolean> | null);
 
-  /**
-   * The real value behind the bitmask instance.
-   */
-  #value: number;
-
-  /**
-   * The structure of valid states and their associated values.
-   */
-  #validStates: Map<string, number>;
-
   /**
    * The enum associated with this structure.
    * @readonly
    */
   states: Record<string, string>;
 
-  /**
-   * Generates the valid states and their associated values.
-   * @param states The structure defining the valid states and their associated values. Defaults to null.
-   */
-  #generateValidStates(states?: Record<string, boolean> | null): void;
-
-  /**
-   * Generates an enum based on the provided valid states.
-   */
-  #generateEnum(): void;
-
-  /**
-   * Calculate the default value of the bitmask based on the initial states
-   * @param initialStates  - The structure defining the valid states and their associated values. Defaults to {}
-   * @returns           The bitmask value
-   */
-  #computeValue(initialStates?: Record<string, boolean>): number;
-
-  /**
-   * Checks a state and throws an error if it doesn't exist.
-   * @param state     - Name of the state to check.
-   */
-  #checkState(state: string): void;
```

Comment:
Could you add back a single truly private methods? I would call it something like `#bitmask: true` if you want. I should probably create a convention.

There's not really a difference to TypeScript between 1 truly private method and 2 but there's a difference between 0 and 1. It's a bit of a complicated explanation but it essence it effects subclassing etc.

Reply:
OK.  I'll go do this in the other PRs, too.